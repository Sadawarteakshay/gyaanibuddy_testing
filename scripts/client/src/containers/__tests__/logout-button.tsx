import React from 'react';
import LogoutButton from '../logout-button';

import { render, renderApollo, cleanup, fireEvent } from '../../test-utils';
import { cache, isLoggedInVar } from '../../cache';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing'

describe('logout button', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('render snapshot', () => {
    const LogoutButtonElement = <LogoutButton />;

    // see https://stackoverflow.com/questions/70805929/how-to-fix-error-usehref-may-be-used-only-in-the-context-of-a-router-compon
    render(LogoutButtonElement, {wrapper: MockedProvider});
    
    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(<MockedProvider>{LogoutButtonElement}</MockedProvider>)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });

  it('renders logout button', async () => {
    renderApollo(<LogoutButton />);
  });

  it('complete logout', async () => {
    isLoggedInVar(true);
    localStorage.setItem('token', window.btoa('testTokenValue'));
    localStorage.setItem('userId', window.btoa('abc123'));
    const { getByTestId } = renderApollo(<LogoutButton />, { cache });
    fireEvent.click(getByTestId('logout-button'));
    expect(isLoggedInVar()).toBeFalsy();
    expect(localStorage.getItem('token')).toBeNull();
    expect(localStorage.getItem('userId')).toBeNull();
  });
});
