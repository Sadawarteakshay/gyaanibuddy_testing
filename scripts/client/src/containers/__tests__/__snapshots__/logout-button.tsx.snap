// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logout button render snapshot 1`] = `
<button
  className="css-qfnldk"
  data-testid="logout-button"
  onClick={[Function]}
>
  <svg>
    exit.svg
  </svg>
  Logout
</button>
`;
